pipenv/cli.py

```diff
@@ -169,47 +203,68 @@ def do_create_virtualenv(three=None):
     do_where(virtualenv=True, bare=False)
 
 
+def get_downloads_info():
+
+    info = []
+
+    for fname in os.listdir(project.download_location):
+        name = fname.split('-')[0]
+
+        # Remove file extensions from name.
+        version = fname.split('-')[1]
+        version = version.replace('.tar.gz', '').replace('.zip', '').replace('.egg', '')
+
+        # Get the hash of each file.
+        c = delegator.run('pip hash {}'.format(os.sep.join([project.download_location, fname])))
+        hash = c.out.split('--hash=')[1].strip()
+
+        info.append(dict(name=name, version=version, hash=hash))
+
+    return info
+
 def do_lock(dev=False):
     """Executes the freeze functionality."""
 
     click.echo(crayons.yellow('Assuring all dependencies from Pipfile are installed...'))
 
     # Purge the virtualenv, for development dependencies.
-    do_purge(bare=True)
+    do_purge(downloads=True)
 
     click.echo(crayons.yellow('Locking {0} dependencies...'.format(crayons.red('[dev-packages]'))))
 
     # Install only development dependencies.
-    do_install_dependencies(dev=True, only=True, bare=True)
+    do_download_dependencies(dev=True, only=True, bare=True)
 
     # Load the Pipfile and generate a lockfile.
     p = pipfile.load(project.pipfile_location)
     lockfile = json.loads(p.lock())
 
     # Pip freeze development dependencies.
-    c = delegator.run('{0} freeze'.format(which_pip()))
+    # c = delegator.run('{0} freeze'.format(which_pip()))
+    results = get_downloads_info()
```

Comment:
I did this in peep—downloading all the packages (or even just one of them; I forget) and then trying to match up the downloads with their package and version after the fact. It was one of the most fertile sources of bugs in the entire project and a major motivator for merging it into pip proper. I clearly don't have a better idea for you, or I would have used it myself. :-)